framework-cloud/framework-openshift-operator/src/main/java/org/kie/cloud/openshift/operator/scenario/KieServerWithDatabaseScenarioImpl.java

```diff
@@ -0,0 +1,155 @@
+    public List<WorkbenchDeployment> getWorkbenchDeployments() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public List<KieServerDeployment> getKieServerDeployments() {
+        return Arrays.asList(kieServerDeployment);
+    }
+
+    @Override
+    public List<SmartRouterDeployment> getSmartRouterDeployments() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public List<ControllerDeployment> getControllerDeployments() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public SsoDeployment getSsoDeployment() {
+        return ssoDeployment;
+    }
+
+    @Override
+    public DatabaseDeployment getDatabaseDeployment() {
+        return databaseDeployment;
+    }
+
+    @Override
+    protected void deployCustomResource() {
+        if (deploySso) {
+            ssoDeployment = SsoDeployer.deploySecure(project);
+            URL ssoSecureUrl = ssoDeployment.getSecureUrl().orElseThrow(() -> new RuntimeException("RH SSO secure URL not found."));
+
+            Sso sso = new Sso();
+            sso.setAdminUser(DeploymentConstants.getSsoServiceUser());
+            sso.setAdminPassword(DeploymentConstants.getSsoServicePassword());
+            sso.setUrl(SsoDeployer.createSsoEnvVariable(ssoSecureUrl.toString()));
+            sso.setRealm(DeploymentConstants.getSsoRealm());
+            sso.setDisableSSLCertValidation(true);
+
+            Auth auth = new Auth();
+            auth.setSso(sso);
+            kieApp.getSpec().setAuth(auth);
+        }
+
+        for (Server server : kieApp.getSpec().getObjects().getServers()) {
+            registerCustomTrustedSecret(server);
+        }
+
+        // deploy application
+        getKieAppClient().create(kieApp);
+        // Wait until the operator reconciliate the KieApp and add there missing informations
+        new SupplierWaiter<KieApp>(() -> getKieAppClient().withName(OpenShiftConstants.getKieApplicationName()).get(), kieApp -> kieApp.getStatus() != null).reason("Waiting for reconciliation to initialize all fields.")
+                                                                                                                                                            .timeout(TimeUnit.MINUTES, 1).waitFor();
+
+        kieServerDeployment = new KieServerOperatorDeployment(project, getKieAppClient());
+        kieServerDeployment.setUsername(DeploymentConstants.getAppUser());
+        kieServerDeployment.setPassword(DeploymentConstants.getAppPassword());
+
+        databaseDeployment = new DatabaseDeploymentImpl(project);
+
+        logger.info("Waiting until all services are created.");
+        try {
+            new SimpleWaiter(() -> kieServerDeployment.isReady()).reason("Waiting for Kie server service to be created.").timeout(TimeUnit.MINUTES, 1).waitFor();
+            new SimpleWaiter(() -> databaseDeployment.isReady()).reason("Waiting for Database service to be created.").timeout(TimeUnit.MINUTES, 1).waitFor();
```

Comment:
Database should be always ready before the KIE Server, is better to switch the waiter order so it can be better visible in exception stack trace what failed.
```suggestion
            new SimpleWaiter(() -> databaseDeployment.isReady()).reason("Waiting for Database service to be created.").timeout(TimeUnit.MINUTES, 1).waitFor();
            new SimpleWaiter(() -> kieServerDeployment.isReady()).reason("Waiting for Kie server service to be created.").timeout(TimeUnit.MINUTES, 1).waitFor();
```

Reply:
Done